File: gts/src/main/java/com/muratkagan/gts/entities/Artist.java
package com.muratkagan.gts.entities;

import java.time.OffsetDateTime;
import java.util.Map;

import jakarta.persistence.*;

@Entity
@Table(name = "artists")
public class Artist {
	
	@Id
	@Column(name = "id")
	private int id;
	
	@Column(name = "name")
	private String name;
	
	@Column(name = "surname")
	private String surname;
	
	@Column(name = "country_id")
	private int countryId;
	
	@Column(name = "bio", columnDefinition = "text")
	private String bio;
	
	@Column(name = "socialLinks", columnDefinition = "jsonb")
	private String socialLinks;
	
	@Column(name = "created_at", insertable = false, updatable = false)
	private OffsetDateTime createdAt;

	@Column(name = "updated_at", insertable = false, updatable = false)
	private OffsetDateTime updatedAt;
	
	// Getters and setters

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getSurname() {
		return surname;
	}

	public void setSurname(String surname) {
		this.surname = surname;
	}

	public int getCountryId() {
		return countryId;
	}

	public void setCountryId(int countryId) {
		this.countryId = countryId;
	}

	public String getBio() {
		return bio;
	}

	public void setBio(String bio) {
		this.bio = bio;
	}

	public String getSocialLinks() {
		return socialLinks;
	}

	public void setSocialLinks(String socialLinks) {
		this.socialLinks = socialLinks;
	}

	public OffsetDateTime getCreatedAt() {
		return createdAt;
	}

	public void setCreatedAt(OffsetDateTime createdAt) {
		this.createdAt = createdAt;
	}

	public OffsetDateTime getUpdatedAt() {
		return updatedAt;
	}

	public void setUpdatedAt(OffsetDateTime updatedAt) {
		this.updatedAt = updatedAt;
	}

}

File: gtsui/dtos/artist.ts
export type ArtistCreateDto = {
  name: string;
  surname: string;
  countryId?: number | null;
  cityId?: number | null;
  bio?: string | null;
  socialLinks?: Map<string, string> | null;
};

export type ArtistUpdateDto = {
  name: string;
  surname: string;
  countryId?: number | null;
  cityId?: number | null;
  bio?: string | null;
  socialLinks?: Map<string, string> | null;
};
export type ArtistResponseDto = {
  name: string;
  surname: string;
  countryId?: number | null;
  cityId?: number | null;
  bio?: string | null;
  socialLinks?: Map<string, string> | null;
  createdAt: string;
  updatedAt: string;
};

export type ArtistListItemDto = {
  name: string;
  surname: string;
  countryId?: number | null;
  cityId?: number | null;
  bio?: string | null;
  socialLinks?: Map<string, string> | null;
  createdAt: string;
  updatedAt: string;
};
